packages/editor/src/browser/editor.ts

```diff
@@ -148,7 +148,33 @@ export interface EditorMouseEvent {
     readonly target: MouseTarget;
 }
 
-export interface TextEditor extends Disposable, TextEditorSelection, Navigatable {
+export const enum EncodingMode {
+
+    /**
+     * Instructs the encoding support to encode the current input with the provided encoding
+     */
+    Encode,
+
+    /**
+     * Instructs the encoding support to decode the current input with the provided encoding
+     */
+    Decode
+}
+
+export interface IEncodingSupport {
```

Comment:
please don't use I prefixes, see https://github.com/theia-ide/theia/wiki/Coding-Guidelines#interfaces

Reply:
Can we inline it into `TextEditor`? I don't see `IEncodingSupport` type is used anywhere.

Reply:
The interface 'IEncodingSupport' comes from VSCode. Not sure if we should rename it or inline it into TextEditor. What do you think?

Reply:
It is fine to modify copied code. Otherwise we will end up with a lot of code not confirmed to our standards. Just add a link that code originally copied and modified from there.

Reply:
Hi @a1994846931931 

> The interface 'IEncodingSupport' comes from VSCode

Please clarify if only the name `IEncodingSupport` was copied from VS Code or also the content of the interface? 

I have not followed closely, but has a CQ been registered to cover the part(s) of this PR that have been copied from somewhere, if applicable?

Reply:
@akosyakov `IEncodingSupport ` is inlined into `TextEditor` now. Could you check it, please?